<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Peter Ralph, University of Oregon">
  <title>The tree sequence:</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="reveal.js/css/reveal.css">
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="reveal.js/css/theme/simple.css" id="theme">
  <!-- Printing and PDF exports -->
  <script>
    var link = document.createElement( 'link' );
    link.rel = 'stylesheet';
    link.type = 'text/css';
    link.href = window.location.search.match( /print-pdf/gi ) ? 'reveal.js/css/print/pdf.css' : 'reveal.js/css/print/paper.css';
    document.getElementsByTagName( 'head' )[0].appendChild( link );
  </script>
  <!--[if lt IE 9]>
  <script src="reveal.js/lib/js/html5shiv.js"></script>
  <![endif]-->
  <style type="text/css">
  
  .reveal { font-size: 30px; }
  
  .reveal h1 { font-size: 1.5em; } 
  
  .reveal h2 { font-size: 1.2em; } 
  
  .reveal .slides { text-align: left; }
  
  .reveal .slides figure { text-align: center; }
  
  .centered p { text-align: center; }
  
  /* apply this to the columns div */
  div.vcentered { display: flex; align-items: center; }
  
  .caption p { font-size: 75%; }
  
  .greyed { color: #4444; }
  
  div.floatright { float: right; text-align: right; }
  
  .reveal figcaption { display: none; }
  
  .reveal img { border: none; }
  
  .reveal section img { border: none; }
  
  .MathJax_Display { margin: 0; }
  
  img.finger { vertical-align: text-top; }
  
  </style>
  <div style="display: none">
  \[
  \newcommand{\E}{\mathbb{E}}
  \renewcommand{\P}{\mathbb{P}}
  \DeclareMathOperator{\var}{var}
  \]
  </div>
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section id="title-slide">
  <h1 class="title">The tree sequence:</h1>
  <p class="subtitle">using genealogies to store, analyze,<br> and interpret large genomic datasets</p>
  <p class="author">Peter Ralph, University of Oregon</p>
  <p class="date">Empirical Research Workshop Series // August 2020</p>
</section>

<section><section id="genomes-and-applications" class="title-slide slide level1"><h1>Genomes, and applications</h1></section><section id="genomes" class="slide level2">
<h2>Genomes</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li class="fragment">are very big (<span class="math inline">\(10^7\)</span>–<span class="math inline">\(10^{12}\)</span> nucleotides)</li>
<li class="fragment">encode the basic <em>mechanisms</em> of life</li>
<li class="fragment">reflect past <em>history</em> and <em>process</em></li>
</ul>
</div><div class="column" style="width:50%;">
<figure>
<img data-src="figs/genome.png" alt="https://www.genome.gov/about-genomics/fact-sheets/A-Brief-Guide-to-Genomics" /><figcaption>https://www.genome.gov/about-genomics/fact-sheets/A-Brief-Guide-to-Genomics</figcaption>
</figure>
</div>
</div>
</section><section id="genomes-1" class="slide level2">
<h2>Genomes</h2>
<div class="columns">
<div class="column" style="width:30%;">
<p><em>For example, in humans (low diversity):</em></p>
<ul>
<li>out of <span class="math inline">\(3 \times 10^8\)</span> nucleotides,</li>
<li>individuals differ at around 5 million sites</li>
<li>88 million varying sites in 2500 genomes <!-- 1000G: https://www.nature.com/articles/nature15393 --></li>
</ul>
</div><div class="column" style="width:70%;">
<div class="centered">
<figure>
<img data-src="figs/genome_cost.png" alt="https://www.genome.gov/about-genomics/fact-sheets/DNA-Sequencing-Costs-Data" /><figcaption>https://www.genome.gov/about-genomics/fact-sheets/DNA-Sequencing-Costs-Data</figcaption>
</figure>
</div>
</div>
</div>
</section><section id="genome-wide-association-studies" class="slide level2">
<h2>Genome-wide association studies</h2>
<p><em>What genetic variants affect disease incidence/severity?</em></p>
<div class="fragment">
<p><em>How?</em></p>
<ol type="1">
<li>Fit a model of <code>severity ~ genotype</code> separately at each site, and</li>
<li>look at sites with significant <span class="math inline">\(p\)</span>-value.</li>
</ol>
</div>
</section><section class="slide level2">

<div class="centered">
<figure>
<img data-src="figs/getting_up_gwas.png" alt="https://twitter.com/SbotGwa/status/1281574037597061120" /><figcaption>https://twitter.com/SbotGwa/status/1281574037597061120</figcaption>
</figure>
<div class="caption">
<p><a href="https://twitter.com/SbotGwa">from SbotGwa</a></p>
</div>
</div>
</section><section class="slide level2">

<div class="centered">
<figure>
<img data-src="figs/gwas_hits.png" alt="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5501872/" /><figcaption>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5501872/</figcaption>
</figure>
</div>
<!--

## Genomic-assisted selection

1. Fit a model of `yield ~ genotype` separately at each site, and
2. add up effect across all sites to get a *polygenic score*, or *breeding value*,
3. use high-scorers as parents of next generation.

![http://mooselab.cropsci.illinois.edu/longterm.html](figs/corn_oil.png)

-->
</section><section id="history-from-genomes" class="slide level2">
<h2>History from genomes</h2>
<div class="columns">
<div class="column" style="width:50%;">
<div class="centered">
<figure>
<img data-src="figs/hominins.jpg" alt="https://www.sciencedirect.com/science/article/pii/S096098221101270X" /><figcaption>https://www.sciencedirect.com/science/article/pii/S096098221101270X</figcaption>
</figure>
<div class="caption">
<p><a href="https://www.sciencedirect.com/science/article/pii/S096098221101270X%5D">Lalueza-Fox &amp; Gilbert</a></p>
</div>
</div>
</div><div class="column" style="width:50%;">
<div class="centered">
<figure>
<img data-src="figs/neanderthal.jpg" alt="https://en.wikipedia.org/wiki/Neanderthal#/media/File:Homo_neanderthalensis_adult_male_-head_model-Smithsonian_Museum_of_Natural_History-_2012-05-17.jpg" style="width:80.0%" /><figcaption>https://en.wikipedia.org/wiki/Neanderthal#/media/File:Homo_neanderthalensis_adult_male_-<em>head_model</em>-<em>Smithsonian_Museum_of_Natural_History</em>-_2012-05-17.jpg</figcaption>
</figure>
<div class="caption">
<p><a href="https://commons.wikimedia.org/w/index.php?curid=20187477">flickr:timevanson</a></p>
</div>
</div>
</div>
</div>
</section><section id="common-tasks" class="slide level2">
<h2>Common tasks:</h2>
<ol type="1">
<li><em>Storing</em> genomes (<span class="math inline">\(10^7\)</span> sequences of <span class="math inline">\(10^7\)</span> nucleotides)</li>
<li><em>Computing</em> similarity, correlation</li>
<li><em>Simulating</em> these data from evolutionary scenarios</li>
<li><em>Reconstructing</em> genealogies</li>
</ol>
</section><section id="section" class="slide level2">
<h2></h2>
<p>A data structure is…</p>
<div class="fragment">
<p><strong>succinct</strong> if it only stores each bit of information once.</p>
</div>
<div class="fragment">
<p><strong>descriptive</strong> if it reflects the underlying process.</p>
</div>
<div class="fragment">
<p><strong>So:</strong> let’s think about the process that generated the data!</p>
</div>
</section></section>
<section><section id="genomes-and-genealogies" class="title-slide slide level1"><h1>Genomes and genealogies</h1></section><section id="meiosis-recombination" class="slide level2">
<h2>Meiosis &amp; Recombination</h2>
<figure>
<img data-src="figs/recombination-cartoon-crop.png" alt="recombination" /><figcaption>recombination</figcaption>
</figure>
<div class="centered">
<p>You have two copies of each chromosome, one from each parent.</p>
</div>
<div class="fragment">
<div class="centered">
<p>When you make a gamete, the copies <em>recombine</em>,</p>
<p>and copying errors lead to <em>mutations</em>.</p>
</div>
</div>
</section><section id="section-1" class="slide level2">
<h2></h2>
<p>Your two copies of the genome were inherited, noisily, from your two parents,</p>
<div class="fragment">
<p>and from your four grandparents,</p>
</div>
<div class="fragment">
<p>and your eight great-grandparents</p>
</div>
<div class="fragment">
<p>and your sixteen great-great-grandparents</p>
</div>
<div class="fragment">
<p><strong>but:</strong> you have not inherited <em>any</em> genome from <em>most</em> of your ancestors, past about 10 generations ago.</p>
</div>
</section><section id="section-2" class="slide level2">
<h2></h2>
<div class="centered">
<p><img data-src="figs/gcbias/parents_grandparents.png" style="width:60.0%" /></p>
<div class="caption">
<p>from <a href="https://gcbias.org/2013/11/11/how-does-your-number-of-genetic-ancestors-grow-back-over-time/">gcbias.org</a></p>
</div>
</div>
</section><section id="section-3" class="slide level2">
<h2></h2>
<div class="centered">
<p><img data-src="figs/gcbias/male_line.png" style="width:60.0%" /></p>
<div class="caption">
<p>from <a href="https://gcbias.org/2013/11/11/how-does-your-number-of-genetic-ancestors-grow-back-over-time/">gcbias.org</a></p>
</div>
</div>
</section><section id="section-4" class="slide level2">
<h2></h2>
<div class="columns vcentered">
<div class="column" style="width:50%;">
<p>In fact, the number of <span class="math inline">\(k^\text{th}\)</span> <em>genetic</em> ancestors grows <em>linearly</em> with <span class="math inline">\(k\)</span>.</p>
</div><div class="column" style="width:50%;">
<div class="centered">
<p><img data-src="figs/gcbias/num_genetics_vs_genealogical_ancs.png" /></p>
<div class="caption">
<p>from <a href="https://gcbias.org/2013/11/11/how-does-your-number-of-genetic-ancestors-grow-back-over-time/">gcbias.org</a></p>
</div>
</div>
</div>
</div>
</section></section>
<section><section id="looking-backwards" class="title-slide slide level1"><h1>Looking backwards</h1></section><section id="section-5" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>Tracing back the ancestry of some chromosomes:</p>
<ul>
<li>blocks labeled by who inherits from them</li>
<li><span class="greyed"> blocks can <em>split</em> </span></li>
<li><span class="greyed"> or <em>coalesce</em>, </span></li>
<li><span class="greyed"> and <em>mutations</em> lead to differences. </span></li>
</ul>
<p><span class="greyed"> Result: a <em>labeled genealogy</em> containing all the <em>genealogical trees</em>. </span></p>
</div><div class="column" style="width:50%;">
<div class="floatright">
<p><img data-src="figs/frag_coal/fragmentation_coalescence.1.png" style="width:80.0%" /></p>
</div>
</div>
</div>
</section><section id="section-6" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>Tracing back the ancestry of some chromosomes:</p>
<ul>
<li>blocks labeled by who inherits from them</li>
<li>blocks can <em>split</em></li>
<li><span class="greyed"> or <em>coalesce</em>, </span></li>
<li><span class="greyed"> and <em>mutations</em> lead to differences. </span></li>
</ul>
<p><span class="greyed"> Result: a <em>labeled genealogy</em> containing all the <em>genealogical trees</em>. </span></p>
</div><div class="column" style="width:50%;">
<div class="floatright">
<p><img data-src="figs/frag_coal/fragmentation_coalescence.2.png" style="width:80.0%" /></p>
</div>
</div>
</div>
</section><section id="section-7" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>Tracing back the ancestry of some chromosomes:</p>
<ul>
<li>blocks labeled by who inherits from them</li>
<li>blocks can <em>split</em></li>
<li>or <em>coalesce</em>,</li>
<li><span class="greyed"> and <em>mutations</em> lead to differences. </span></li>
</ul>
<p><span class="greyed"> Result: a <em>labeled genealogy</em> containing all the <em>genealogical trees</em>. </span></p>
</div><div class="column" style="width:50%;">
<div class="floatright">
<p><img data-src="figs/frag_coal/fragmentation_coalescence.3.png" style="width:80.0%" /></p>
</div>
</div>
</div>
</section><section id="section-8" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>Tracing back the ancestry of some chromosomes:</p>
<ul>
<li>blocks labeled by who inherits from them</li>
<li>blocks can <em>split</em></li>
<li>or <em>coalesce</em>,</li>
<li><span class="greyed"> and <em>mutations</em> lead to differences. </span></li>
</ul>
<p><span class="greyed"> Result: a <em>labeled genealogy</em> containing all the <em>genealogical trees</em>. </span></p>
</div><div class="column" style="width:50%;">
<div class="floatright">
<p><img data-src="figs/frag_coal/fragmentation_coalescence.4.png" style="width:80.0%" /></p>
</div>
</div>
</div>
</section><section id="section-9" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>Tracing back the ancestry of some chromosomes:</p>
<ul>
<li>blocks labeled by who inherits from them</li>
<li>blocks can <em>split</em></li>
<li>or <em>coalesce</em>,</li>
<li><span class="greyed"> and <em>mutations</em> lead to differences. </span></li>
</ul>
<p><span class="greyed"> Result: a <em>labeled genealogy</em> containing all the <em>genealogical trees</em>. </span></p>
</div><div class="column" style="width:50%;">
<div class="floatright">
<p><img data-src="figs/frag_coal/fragmentation_coalescence.5.png" style="width:80.0%" /></p>
</div>
</div>
</div>
</section><section id="section-10" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>Tracing back the ancestry of some chromosomes:</p>
<ul>
<li>blocks labeled by who inherits from them</li>
<li>blocks can <em>split</em></li>
<li>or <em>coalesce</em>,</li>
<li><span class="greyed"> and <em>mutations</em> lead to differences. </span></li>
</ul>
<p><span class="greyed"> Result: a <em>labeled genealogy</em> containing all the <em>genealogical trees</em>. </span></p>
</div><div class="column" style="width:50%;">
<div class="floatright">
<p><img data-src="figs/frag_coal/fragmentation_coalescence.6.png" style="width:80.0%" /></p>
</div>
</div>
</div>
</section><section id="section-11" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>Tracing back the ancestry of some chromosomes:</p>
<ul>
<li>blocks labeled by who inherits from them</li>
<li>blocks can <em>split</em></li>
<li>or <em>coalesce</em>,</li>
<li>and <em>mutations</em> lead to differences.</li>
</ul>
<p><span class="greyed"> Result: a <em>labeled genealogy</em> containing all the <em>genealogical trees</em>. </span></p>
</div><div class="column" style="width:50%;">
<div class="floatright">
<p><img data-src="figs/frag_coal/fragmentation_coalescence.7.png" style="width:80.0%" /></p>
</div>
</div>
</div>
</section><section id="section-12" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>Tracing back the ancestry of some chromosomes:</p>
<ul>
<li>blocks labeled by who inherits from them</li>
<li>blocks can <em>split</em></li>
<li>or <em>coalesce</em>,</li>
<li>and <em>mutations</em> lead to differences.</li>
</ul>
<p>Result: a <em>labeled genealogy</em> containing all the <em>genealogical trees</em>.</p>
</div><div class="column" style="width:50%;">
<div class="floatright">
<p><img data-src="figs/frag_coal/fragmentation_coalescence.7.png" style="width:80.0%" /></p>
</div>
</div>
</div>
<!-- explain tree sequences and why they are so efficient (5min) -->
</section></section>
<section><section id="the-tree-sequence" class="title-slide slide level1"><h1>The tree sequence</h1></section><section id="history-is-a-sequence-of-trees" class="slide level2">
<h2>History is a sequence of trees</h2>
<p>For a set of sampled chromosomes, at each position along the genome there is a genealogical tree that says how they are related.</p>
<div class="fragment">
<figure>
<img data-src="figs/ts_ex/sim_ts.anim.gif" alt="Trees along a chromosome" style="width:100.0%" /><figcaption>Trees along a chromosome</figcaption>
</figure>
</div>
</section><section class="slide level2">

<p>The <strong>succinct tree sequence</strong></p>
<div class="floatright">
<p>is a way to succinctly describe this, er, sequence of trees</p>
<p><em>and</em> the resulting genome sequences.</p>
</div>
<div class="fragment">
<div class="floatright">
<figure>
<img data-src="figs/jerome.jpeg" alt="jerome kelleher" style="width:50.0%" /><figcaption>jerome kelleher</figcaption>
</figure>
<div class="caption">
<p>by Jerome Kelleher, in <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004842">Kelleher, Etheridge, and McVean</a></p>
</div>
</div>
</div>
</section><section id="example-three-samples-two-trees-two-variant-sites" class="slide level2">
<h2>Example: three samples; two trees; two variant sites</h2>
<figure>
<img data-src="figs/example_tree_sequence.png" alt="Example tree sequence" /><figcaption>Example tree sequence</figcaption>
</figure>
</section><section id="nodes-and-edges" class="slide level2">
<h2>Nodes and edges</h2>
<dl>
<dt>Edges</dt>
<dd><p>Who inherits from who.</p>
<p>Records: interval (left, right); parent node; child node.</p>
</dd>
<dt>Nodes</dt>
<dd><p>The ancestors those happen in.</p>
<p>Records: time ago (of birth); ID (implicit).</p>
</dd>
</dl>
</section><section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.0.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.1.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.2.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.3.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.4.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.5.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.6.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section id="sites-and-mutations" class="slide level2">
<h2>Sites and mutations</h2>
<dl>
<dt>Mutations</dt>
<dd><p>When state changes along the tree.</p>
<p>Records: site it occured at; node it occurred in; derived state.</p>
</dd>
<dt>Sites</dt>
<dd><p>Where mutations fall on the genome.</p>
<p>Records: genomic position; ancestral (root) state; ID (implicit).</p>
</dd>
</dl>
</section><section class="slide level2">

<figure>
<img data-src="sites_muts_walkthrough/sites_muts_walkthrough.0.png" alt="Adding mutations" /><figcaption>Adding mutations</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="sites_muts_walkthrough/sites_muts_walkthrough.1.png" alt="Adding mutations" /><figcaption>Adding mutations</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="sites_muts_walkthrough/sites_muts_walkthrough.2.png" alt="Adding mutations" /><figcaption>Adding mutations</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="sites_muts_walkthrough/sites_muts_walkthrough.3.png" alt="Adding mutations" /><figcaption>Adding mutations</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="sites_muts_walkthrough/sites_muts_walkthrough.4.png" alt="Adding mutations" /><figcaption>Adding mutations</figcaption>
</figure>
</section><section class="slide level2">

<p><strong>The result:</strong> an encoding of the genomes <em>and</em> all the genealogical trees.</p>
<div class="centered">
<figure>
<img data-src="figs/example_tree_sequence.png" alt="Example tree sequence" /><figcaption>Example tree sequence</figcaption>
</figure>
</div>
</section></section>
<section><section id="hows-it-work" class="title-slide slide level1"><h1>How’s it work?</h1></section><section id="file-sizes" class="slide level2">
<h2>File sizes</h2>
<div class="centered">
<figure>
<img data-src="figs/tsinfer_sizes.png" alt="file sizes" style="width:90.0%" /><figcaption>file sizes</figcaption>
</figure>
</div>
<div class="caption">
<p>100Mb chromosomes; from <a href="https://www.nature.com/articles/s41588-019-0483-y">Kelleher et al 2018, <em>Inferring whole-genome histories in large population datasets</em></a>, Nature Genetics</p>
</div>
<!-- Estimated sizes of files required to store the genetic variation data for a
simulated human-like chromosome (100 megabases) for up to 10 billion haploid
(5 billion diploid) samples. Simulations were run for 10 1 up to 10 7 haplotypes
using msprime [Kelleher et al., 2016], and the sizes of the resulting files plotted
(points). -->
</section><section class="slide level2">

<figure>
<img data-src="figs/ts_ex/tree_sequence_genotypes.png" alt="genotypes" /><figcaption>genotypes</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/ts_ex/tree_sequence_genotype_and_tree.png" alt="genotypes and a tree" /><figcaption>genotypes and a tree</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/ts_ex/tree_sequence_next_genotype_and_tree.png" alt="genotypes and the next tree" /><figcaption>genotypes and the next tree</figcaption>
</figure>
</section><section id="for-n-samples-genotyped-at-m-sites" class="slide level2">
<h2>For <span class="math inline">\(N\)</span> samples genotyped at <span class="math inline">\(M\)</span> sites</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><em>Genotype matrix</em>:</p>
<p><span class="math inline">\(N \times M\)</span> things.</p>
</div><div class="column" style="width:50%;">
<p><em>Tree sequence:</em></p>
<ul>
<li><span class="math inline">\(2N-2\)</span> edges for the first tree</li>
<li><span class="math inline">\(\sim 4\)</span> edges per each of <span class="math inline">\(T\)</span> trees</li>
<li><span class="math inline">\(M\)</span> mutations</li>
</ul>
<p><span class="math inline">\(O(N + T + M)\)</span> things</p>
</div>
</div>
<figure>
<img data-src="figs/ts_ex/tree_sequence_genotype_and_tree.png" alt="genotypes and a tree" style="width:70.0%" /><figcaption>genotypes and a tree</figcaption>
</figure>
</section><section id="fast-genotype-statistics" class="slide level2">
<h2>Fast genotype statistics</h2>
<div class="centered">
<figure>
<img data-src="figs/treestats/benchmarks_without_copy_longer_genome.png" alt="efficiency of treestat computation" style="width:70.0%" /><figcaption>efficiency of treestat computation</figcaption>
</figure>
</div>
<div class="caption">
<p>from <a href="https://www.biorxiv.org/content/10.1101/779132v1?rss=1">R., Thornton and Kelleher 2019, <em>Efficiently summarizing relationships in large samples</em></a>, bioRxiv</p>
</div>
</section></section>
<section><section id="demonstration" class="title-slide slide level1"><h1>Demonstration</h1></section><section class="slide level2">

<p><a href="https://github.com/petrelharp/reed_aug_2020/blob/master/ts-demo.ipynb">jupyter demonstration</a></p>
</section></section>
<section><section id="wrap-up" class="title-slide slide level1"><h1>Wrap up</h1></section><section id="code-availability" class="slide level2">
<h2>Code availability</h2>
<div class="columns">
<div class="column">
<p>Everything is efficient, open, and tested.</p>
<figure>
<img data-src="figs/tskit_logo.png" alt="tskit logo" style="width:100.0%" /><figcaption>tskit logo</figcaption>
</figure>
</div><div class="column">
<ul>
<li><p><code>tskit</code>: tree sequence tools</p>
<p><a href="https://github.com/tskit-dev/tskit">github:tskit-dev/tskit</a></p></li>
<li><p><code>stdpopsim</code>: a library of “standard” simulation tools</p>
<p><a href="https://github.com/popsim-consortium/stdpopsim">github:popsim-consortium/stdpopsim</a></p></li>
<li><p><code>msprime</code>: coalescent simulator,</p>
<p><a href="https://github.com/tskit-dev/msprime">github:tskit-dev/msprime</a></p></li>
<li><p><code>SLiM</code> : forwards evolutionary simulator</p>
<p><a href="https://messerlab.org/slim/">messerlab.org/slim</a></p></li>
</ul>
</div>
</div>
</section><section id="thanks" class="slide level2">
<h2>Thanks</h2>
<div class="columns">
<div class="column">
<ul>
<li>Philipp Messer</li>
<li>Boyana Norris</li>
<li>Gil McVean</li>
<li>Simon Myers</li>
<li>Wilder Wohns</li>
<li>Kevin Thornton</li>
<li>Graham Coop</li>
<li>Gideon Bradburd</li>
<li>Alison Etheridge</li>
<li>CJ Battey</li>
<li>Andy Kern</li>
</ul>
</div><div class="column">
<ul>
<li>Jerome Kelleher</li>
<li>Yan Wong</li>
<li>Ben Jeffery</li>
<li>Ben Haller</li>
<li>Georgia Tsambos</li>
<li>Jared Galloway</li>
<li><code>tskit-dev</code> and <code>popsim-consortium</code></li>
</ul>
<p>Funding: NSF, NIH, UO (PR); Wellcome Trust (JK);</p>
<p>Slides with <a href="http://hakim.se">reveal.js</a> and <a href="https://pandoc.org/">pandoc</a>.</p>
<figure>
<img data-src="figs/tskit_logo.png" alt="tskit logo" style="width:50.0%" /><figcaption>tskit logo</figcaption>
</figure>
</div>
</div>
<!-- questions?? -->
</section><section id="section-13" class="slide level2" data-background-image="figs/guillemots_thanks.png" data-background-position="center" data-background-size="50%">
<h2></h2>
</section></section>
    </div>
  </div>

  <script src="reveal.js/lib/js/head.min.js"></script>
  <script src="reveal.js/js/reveal.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        // Display the page number of the current slide
        slideNumber: true,
        // Push each slide change to the browser history
        history: true,
        // Transition style
        transition: 'none', // none/fade/slide/convex/concave/zoom
        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // Optional reveal.js plugins
        dependencies: [
          { src: 'reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
          { src: 'reveal.js/plugin/zoom-js/zoom.js', async: true },
          { src: 'reveal.js/plugin/math/math.js', async: true },
          { src: 'reveal.js/plugin/notes/notes.js', async: true }
        ]
      });
    </script>
    </body>
</html>
